File: meldingsvarsler-fasit-forslag/src/main/java/no/sonat/meldingsvarsler/sms/SMSAbonnent.java
package no.sonat.meldingsvarsler.sms;

import no.sonat.meldingsvarsler.abonnent.Abonnent;
import no.sonat.meldingsvarsler.Melding;

public class SMSAbonnent implements Abonnent {
    private final String navn;
    private final String telefonnummer;

    public SMSAbonnent(String navn, String telefonnummer) {
        this.navn = navn;
        this.telefonnummer = telefonnummer;
    }

    @Override
    public boolean abonnererPaa(Melding melding) {
        return melding instanceof SMSMelding;
    }

    @Override
    public String navn() {
        return navn;
    }

    @Override
    public Boolean erAktiv() {
        return true;
    }

    public String telefonnummer() {
        return telefonnummer;
    }
}

File: meldingsvarsler-fasit-forslag/App/SMSAbonnent.cs
namespace App {
    public class SmsAbonnent : IAbonnent
    {
        public SmsAbonnent(string navn, string telefonNummer)
        {
            Navn = navn;
            TelefonNummer = telefonNummer;
        }


        public bool AbonnererPaa(IMelding melding)
        {
            return melding is SmsMelding;
        }


        public string Navn { get; }

        public bool ErAktiv => true;

        public string TelefonNummer { get; }
    }
}